dts/arm/st/wba/stm32wba.dtsi

```diff
@@ -98,6 +98,12 @@
 		};
 	};
 
+	jtag_port: jtag_port {
+		compatible = "gpio-jtag";
+		jtag-gpios = <&gpioa 13 0>, <&gpioa 14 0>, <&gpioa 15 0>,
+			     <&gpiob 3 0> ,<&gpiob 4 0>;
+	};
```

Comment:
I'm not sure the "gpio-jtag" is a well-defined binding. The fact you're using GPIO here is because you want to use the Zephyr GPIO API, not because this necessarily describes the hardware well. In fact, I find it a bit confusing as I have no clue by reading devicetree what all those pins mean. Which one is TDI, TMS...?

I'd suggest exploring a solution using pinctrl, would make things easier as well (technically you do not need a driver to use it, just use `PINCTRL_DT_DEFINE(DT_NODELABEL(jtag_port))`.

Reply:
> I'd suggest exploring a solution using pinctrl, would make things easier as well

Indeed. See my note in the PR description. Thanks for confirming my doubt.

> In fact, I find it a bit confusing as I have no clue by reading devicetree what all those pins mean. Which one is TDI, TMS...?

Tbh, if they are not used otherwise, I'm not sure this is useful. Note that even with pinctrl, it is difficult to extract one single pin. So being able to identify them is nice but, I don't see how it could be used.

> (technically you do not need a driver to use it, just use PINCTRL_DT_DEFINE(DT_NODELABEL(jtag_port)).

Yes, but I was wondering where to actually use it. Maybe I can just reuse the same bootstrap that I'm using currently with GPIO in the end.

Reply:
I guess you could have something like:


```
jtag: connector {
    compatible = "jtag-connector";
    pinctrl-0 = <&jtag_tms_pa13...>;
    pinctrl-1 = <&analog_pa13...>;
    pinctrl-names = "default", "sleep";
};
```

Reply:
I've started having a look. Regarding AF, no issue, script handles it quite easily.
Though, each pin has its own specific configuration (speed, pupdr), which is not documented in database.
So, we have 2 options:
- Fill speed and state based on ref man information specifically for each pin (hopefully configuration doesn't depend on series). Gently tedious but doable thanks to the nice script we have but I've heard some are not quite happy with filling pin properties in -pcintrl.dtsi files ;)
- Don't fill the the pin properties, assuming they won't be used (ie, no dynamic switching from analog back to functional debugging state > I wonder about the interest of this use case anyway).

Reply:
> I've started having a look. Regarding AF, no issue, script handles it quite easily. Though, each pin has its own specific configuration (speed, pupdr), which is not documented in database. So, we have 2 options:
> 
> * Fill speed and state based on ref man information specifically for each pin (hopefully configuration doesn't depend on series). Gently tedious but doable thanks to the nice script we have but I've heard some are not quite happy with filling pin properties in -pcintrl.dtsi files ;)

That script, in retrospective, was a mistake that propagated from the very early board-based pinctrl model. The solution implemented for GD32 is far better and more flexible (it's a copy of Linux STM32 model), the script simply generates all AFs, no need to worry if new peripherals are added, all definitions will already be there. Properties are up to the user to figure out, same as you'd do developing with the HAL in bare-metal. I'd suggest reviewing the ST model to at least drop the pin properties from the generated files if this is a problem.

Reply:
> That script, in retrospective, [...]

I didn't wanted to re-open the discussion on this. On my side I'm still not convinced about the scalability of this model. 

> it's a copy of Linux STM32 model

Discussing with Linux STM32 folks, they agree current Zephyr solution is much more suited to our "lot of SoCs, lot of pin combinations" model, and some would even prefer to use this over Linux model (not saying it will happen any day).

> "HAL in bare-metal" 

HAL is not a model of scalability. Besides, from the reports I get from users, I'm not convinced either that STM32 Linux model would be more easily handled than current definitions. There were some issues reported but none that I can remember about the fact of having pin properties being defined in -pinctrl.dtsi files

I can understand that this model is not conceptually perfect, but as long as it will be easy to maintain and there aren't significant issues directly linked to it, I don't see reasons to change. If it happen one day that, for instance, dts pin config is generated from a tool making usage and maintenance eased by a different model, I'll be open to discussion.

Coming back to current issue, I don't see how STM32 Linux model would be of any help.
For now I'll go with solution 2, adding a warning on the fact that it is not designed to swap pins dynamically (but can be updated in board if required). This can be updated later if required.

(Damn, for someone not willing to enter the discussion, I've failed :-) )

Reply:
I guess Linux folks just miss the pre-defined pin combinations, getting AFs from datasheet can be painful. The properties are usually trivial to guess (e.g. pull-ups, open-drain, drive, etc.). So the GD32 is the Linux STM32 model, but better in the sense that you don't need to look for AFs.

Anyway, I won't insist on this, in the end it's there for historical reasons. But no other vendor has been allowed to do the same because it often leads to issues or misconfigurations (e.g. ST still hasn't fixed the I2C case).